src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -837,33 +838,110 @@ subroutine init_atm_static(mesh, dims, configs)
                             messageType=MPAS_LOG_CRIT)
     endif
 
+    call mpas_pool_get_config(mgr % pool, 'tile_bdr', tile_bdr)
+    call mpas_pool_get_config(mgr % pool, 'tile_x', tile_nx)
+    call mpas_pool_get_config(mgr % pool, 'tile_y', tile_ny)
+    call mpas_pool_get_config(mgr % pool, 'missing_value', missing_value)
+
+    allocate(nhs(nCells))
     snoalb(:) = 0.0
+    nhs(:) = 0
+    fillval = 0.0
+    missing_value = real(missing_value, kind=R4KIND) * real(0.01, kind=R4KIND)
 
     do iCell = 1, nCells
+        if (nhs(iCell) == 0) then
+            tile => null()
+            ierr = mgr % get_tile(latCell(iCell), lonCell(iCell), tile)
+            if (ierr /= 0 .or. .not. associated(tile)) then
+                call mpas_log_write('Could not get tile that contained cell $i', intArgs=(/iCell/), messageType=MPAS_LOG_CRIT)
+            end if
 
-       tile => null()
-       ierr = mgr % get_tile(latCell(iCell), lonCell(iCell), tile)
-       if (ierr /= 0 .or. .not. associated(tile)) then
-           call mpas_log_write('Could not get the tile that contained cell $i', intArgs=(/iCell/), messageType=MPAS_LOG_CRIT)
-       end if
+            ierr = mgr % push_tile(tile)
+            if (ierr /= 0) then
+                call mpas_log_write("Error pushing this tile onto the stack: "//trim(tile%fname), messageType=MPAS_LOG_CRIT)
+            end if
+        end if
 
-       if (tile % is_processed) then
-           cycle
-       end if
+        do while (.not. mgr % is_stack_empty())
+            tile => mgr % pop_tile()
 
-       tile % is_processed = .True.
+            if (tile % is_processed) then
+                cycle
+            end if
 
-       ierr = mgr % push_tile(tile)
-       if (ierr /= 0) then
-           call mpas_log_write('Error pushing this tile onto the stack: '//trim(tile%fname), messageType=MPAS_LOG_CRIT)
-       end if
+            call mpas_log_write('Processing tile: '//trim(tile % fname))
+
+            all_pixels_mapped_to_halo_cells = .true.
+
+            do i = supersample_fac * tile_bdr + 1, supersample_fac * (tile_ny + tile_bdr), 1
+                do j = supersample_fac * tile_bdr + 1, supersample_fac * (tile_nx + tile_bdr), 1
```

Comment:
I think these loops need to be flipped i.e.:
```
 do j = supersample_fac * tile_bdr + 1, supersample_fac * (tile_ny + tile_bdr), 1
                do i = supersample_fac * tile_bdr + 1, supersample_fac * (tile_nx + tile_bdr), 1
```

Reply:
I agree.

Reply:
This has been updated.